plaintext-secrets

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Check if yq is installed
+if ! command -v yq &> /dev/null; then
+    echo "Error: yq is not installed."
+    exit 1
+fi
+
+# Function to redact half of the value with asterisks
+redact_value() {
+  local value="$1"
+  local length=${#value}
+  local half_length=$((length / 2))
+  local redacted_value="${value:0:half_length}$(printf '*%.0s' $(seq 1 $half_length))"
+  echo "$redacted_value"
+}
+
+declare passedFilesCount=0
+declare failedFilesCount=0
+declare return_code=0
+
+# Iterate over files matching the naming pattern "helm-values.yaml"
+while IFS= read -r file; do
+  # Parse the values file and detect non-empty string values for keys named "password"
+  plaintext_password=$(yq eval '.. | select(has("password") and .password != "" and (.password | type == "!!str") and (.password | line_comment | . != "ignore-plaintext-pwd-check")) | .password' "$file")
+  plaintext_rootPassword=$(yq eval '.. | select(has("rootPassword") and .rootPassword != "" and (.rootPassword | type == "!!str") and (.rootPassword | line_comment | . != "ignore-plaintext-pwd-check")) | .rootPassword' "$file")
+  plaintext_adminPassword=$(yq eval '.. | select(has("adminPassword") and .adminPassword != "" and (.adminPassword | type == "!!str") and (.adminPassword | line_comment | . != "ignore-plaintext-pwd-check")) | .adminPassword' "$file")
+  plaintext_postgresqlPassword=$(yq eval '.. | select(has("postgresqlPassword") and .postgresqlPassword != "" and (.postgresqlPassword | type == "!!str") and (.postgresqlPassword | line_comment | . != "ignore-plaintext-pwd-check")) | .postgresqlPassword' "$file")
```

Comment:
Feel like its repetative. Could we consider having an array to store keys ("password" "rootPassword" "adminPassword" "postgresqlPassword") and iterate over them?

Reply:
I did try an array to begin with, but the issue then is if any of those password types returns more than one value for a file, we'd need a multi-dimensional array and bash really doesn't handle those well. If it starts becoming an issue, we could always rewrite the script into python to get better support for it

Reply:
Fixed in latest commit